### DISCUSSION TITLE
Access floor db from native code

### DISCUSSION THREAD

**Discussion by TechInjectIndia** [vitusortner/floor#550] [Q&A]
I have a ready flutter app and I want to add some native android activities in the code base, is there a way to access floor DB from a native activity using room library ( [https://developer.android.com/jetpack/androidx/releases/room?gclid=CjwKCAjw7J6EBhBDEiwA5UUM2kJo_AEkjpO4DUp8amAXiTOO527v_pftw7VtgCr_iQlpDxheIy45VhoCIDQQAvD_BwE&gclsrc=aw.ds](url) ) 
NEED Help

**Comment by mqus** [comment]
See #493. TLDR: I think this is not possible(and even if it is, has many downsides) and you would be better off deciding if you use room OR floor (or another Flutter ORM) and do the adapters/abstractions in the application layer.

**Reply by wangmir** [reply]
> See #493. TLDR: I think this is not possible(and even if it is, has many downsides) and you would be better off deciding if you use room OR floor (or another Flutter ORM) and do the adapters/abstractions in the application layer.

I also have this problem right now. I saw #493 too, but anyone has a better solution (not only the functional support from) on sharing the database between native android and flutter side?

I only use read-only database fetched from the server, so I doubly downloaded the database from the flutter and android sides (with a similar code of floor and room, and retrofit of each side). But, it seems not good actually, so I need a better solution.

Can I make a in-memory database from file-backed database from floor? In that case, at least I can try something like,
1. fetch data from server and store-it at the file-backed DB.
2. flutter and android side floor / room will fetch file-backed DB with in-memory database.
